ui/apps/platform/src/Containers/NetworkGraph/types/topology.type.ts

```diff
@@ -40,6 +40,13 @@ export type NamespaceData = {
 
 export type NetworkPolicyState = 'none' | 'both' | 'ingress' | 'egress';
 
+export type NodeDataType =
+    | 'DEPLOYMENT'
+    | 'EXTERNAL_GROUP'
+    | 'EXTERNAL_ENTITIES'
+    | 'CIDR_BLOCK'
+    | 'EXTRANEOUS';
+
```

Comment:
not sure what the best way to have this exist along side the actual `type` strings in the specific data types of each kind of node below. open to suggestions!

Reply:
This is fine. I don't know if there is a fancy Typescript way of going through the `CustomNodeData` type and extracting the value for the `type` field similarly. Seems like overkill anyways

Reply:
Could you use the TS `as const` pattern where the actual constants are exported to get both "for the price of one"?
```suggestion
export const type = {
    DEPLOYMENT: 'DEPLOYMENT',
    EXTERNAL_GROUP: 'EXTERNAL_GROUP',
    EXTERNAL_ENTITIES: 'EXTERNAL_ENTITIES',
    CIDR_BLOCK: 'CIDR_BLOCK',
    EXTRANEOUS: 'EXTRANEOUS',
} as const;
```

Reply:
thanks, will explore this in a later PR